zebra-rpc/src/methods.rs

```diff
@@ -182,14 +191,26 @@ where
     ) -> Self
     where
         Version: ToString,
+        <Mempool as Service<mempool::Request>>::Future: Send,
+        <State as Service<zebra_state::ReadRequest>>::Future: Send,
     {
-        RpcImpl {
+        let runner = Queue::start();
+
+        let rpc_impl = RpcImpl {
             app_version: app_version.to_string(),
-            mempool,
-            state,
-            latest_chain_tip,
+            mempool: mempool.clone(),
+            state: state.clone(),
+            latest_chain_tip: latest_chain_tip.clone(),
             network,
-        }
+            queue_sender: runner.sender(),
+        };
+
+        // run the process queue
+        tokio::spawn(async move {
+            runner.run(mempool, state, latest_chain_tip, network).await;
+        });
```

Comment:
I added this at https://github.com/ZcashFoundation/zebra/pull/4015/commits/12ce44932988ec5d4236c000ede04a3b7899f316 but please check it. I am unsure if this is enough and the right thing to do. Thanks.

Reply:
I'm not sure commit 12ce44932988ec5d4236c000ede04a3b7899f316 will work, because the second task will hide the panic from the first task.

If we return the `JoinHandle` all the way back to the `start` task, it can check it regularly, and forward any panics to the main task. It will look a lot like this:

https://github.com/ZcashFoundation/zebra/blob/13c216cd3c95d10cae4d2cdb31d3fe417b7a1032/zebrad/src/commands/start.rs#L236-L241

Reply:
I am not following the idea, will you like to give it a try ?

Reply:
Sure, I can do that later today

Reply:
https://github.com/ZcashFoundation/zebra/pull/4015/commits/ecfb7edb05922ad2d77a13bc32a3d55f16b7377b